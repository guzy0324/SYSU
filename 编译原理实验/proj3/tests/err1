/* Test case 1: duplicate declarations */
program err1;
class myTest
{
	declarations
		int x, y;
		int x;
	enddeclarations
	method void main()
	{
	/* nothing */	
	}
	method int func(int m, n)
	declarations
	int m;
	enddeclarations
	{
	/* nothing */	
	}
}
