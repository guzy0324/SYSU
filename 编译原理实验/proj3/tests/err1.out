Semantic Error--line: 20, symbol x: redeclared.
Semantic Error--line: 20, symbol m: redeclared.
********************************Symbol Table************************************

          Name Nest-Level  Tree-Node Predefined        Kind      Type      Value     Offset  Dimension     Argnum

  1     myTest          0                            class                                                       
  2          y          1                         variable  type-tree                                            
  3          x          1                         variable  type-tree                                            
  4       main          1  proc-tree             procedure                                                      0
  5       func          1  proc-tree              function  type-tree                                           2
  6          m          2                          ref_arg  type-tree                                            
  7          n          2                          ref_arg  type-tree                                            
************* SYNTAX TREE PRINTOUT ***********

  +-[DUMMYnode]
R-[ProgramOp]
  |   +-[STNode,1,"myTest"]
  | +-[ClassDefOp]
  | | |       +-[DUMMYnode]
  | | |     +-[StmtOp]
  | | |     | +-[DUMMYnode]
  | | |   +-[BodyOp]
  | | |   | |       +-[DUMMYnode]
  | | |   | |     +-[CommaOp]
  | | |   | |     | | +-[DUMMYnode]
  | | |   | |     | +-[TypeIdOp]
  | | |   | |     |   +-[INTEGERTNode]
  | | |   | |   +-[CommaOp]
  | | |   | |   | +-[IDNode,26,"m"]
  | | |   | | +-[DeclOp]
  | | |   | | | +-[DUMMYnode]
  | | |   | +-[BodyOp]
  | | |   |   +-[DUMMYnode]
  | | | +-[MethodOp]
  | | | | |     +-[DUMMYnode]
  | | | | |   +-[TypeIdOp]
  | | | | |   | +-[INTEGERTNode]
  | | | | | +-[SpecOp]
  | | | | | | |   +-[DUMMYnode]
  | | | | | | | +-[RArgTypeOp]
  | | | | | | | | | +-[INTEGERTNode]
  | | | | | | | | +-[CommaOp]
  | | | | | | | |   +-[STNode,7,"n"]
  | | | | | | +-[RArgTypeOp]
  | | | | | |   | +-[INTEGERTNode]
  | | | | | |   +-[CommaOp]
  | | | | | |     +-[STNode,6,"m"]
  | | | | +-[HeadOp]
  | | | |   +-[STNode,5,"func"]
  | | +-[BodyOp]
  | |   |       +-[DUMMYnode]
  | |   |     +-[StmtOp]
  | |   |     | +-[DUMMYnode]
  | |   |   +-[BodyOp]
  | |   |   | +-[DUMMYnode]
  | |   | +-[MethodOp]
  | |   | | |   +-[DUMMYnode]
  | |   | | | +-[SpecOp]
  | |   | | | | +-[DUMMYnode]
  | |   | | +-[HeadOp]
  | |   | |   +-[STNode,4,"main"]
  | |   +-[BodyOp]
  | |     |       +-[DUMMYnode]
  | |     |     +-[CommaOp]
  | |     |     | | +-[DUMMYnode]
  | |     |     | +-[TypeIdOp]
  | |     |     |   +-[INTEGERTNode]
  | |     |   +-[CommaOp]
  | |     |   | +-[IDNode,12,"x"]
  | |     | +-[DeclOp]
  | |     | | +-[DUMMYnode]
  | |     +-[BodyOp]
  | |       |       +-[DUMMYnode]
  | |       |     +-[CommaOp]
  | |       |     | | +-[DUMMYnode]
  | |       |     | +-[TypeIdOp]
  | |       |     |   +-[INTEGERTNode]
  | |       |   +-[CommaOp]
  | |       |   | +-[STNode,2,"y"]
  | |       | +-[DeclOp]
  | |       | | |     +-[DUMMYnode]
  | |       | | |   +-[CommaOp]
  | |       | | |   | | +-[DUMMYnode]
  | |       | | |   | +-[TypeIdOp]
  | |       | | |   |   +-[INTEGERTNode]
  | |       | | | +-[CommaOp]
  | |       | | | | +-[STNode,3,"x"]
  | |       | | +-[DeclOp]
  | |       | |   +-[DUMMYnode]
  | |       +-[BodyOp]
  | |         +-[DUMMYnode]
  +-[ClassOp]
    +-[DUMMYnode]
